schema/deploy/shipping/views.sql

```diff
@@ -131,4 +131,40 @@ create or replace view shipping.flu_assembly_jobs_v1 as
 comment on view shipping.flu_assembly_jobs_v1 is
     'View of flu jobs that still need to be run through the assembly pipeline';
 
+
+create or replace view shipping.return_results_v1 as
+
+    select barcode,
+           case
+             when sample_id is null then 'notReceived'
+             when sample_id is not null and string_agg(present::text, ',') is null then 'processing'
+             when string_agg(present::text, ',') is not null then 'complete'
+           end as status,
+           array_remove(array_agg(
+             distinct
+             case
+                 -- We only return the top level organisms for results so we want to omit subtypes
+                 when present then subpath(organism, 0, 1)::text
+             end), null)  as organisms_present,
+           array_remove(array_agg(
+             distinct
+             case
+                 when not present then subpath(organism, 0, 1)::text
+             end), null) as organisms_absent
```

Comment:
Something like the below reads a bit more clearly to me by making use of aggregate `filter` clause to accomplish the same thing as the `case` and `array_remove`.

```sql
array_agg(distinct subpath(organism, 0, 1)::text)
    filter (where present and organism is not null)
    as organisms_present,

array_agg(distinct subpath(organism, 0, 1)::text)
    filter (where not present and organism is not null)
    as organisms_absent
```